src/scaffoldmaker/meshtypes/meshtype_3d_bladderurethra1.py

```diff
@@ -0,0 +1,879 @@
+                options[key] = 1
+        if options['Number of elements through wall'] > 1:
+            if options['Include ureter']:
+                options['Number of elements through wall'] = 1
+
+    @classmethod
+    def updateSubScaffoldOptions(cls, options):
+        '''
+        Update ostium sub-scaffold options which depend on parent options.
+        '''
+        bladderWallThickness = options['Wall thickness']
+        ostiumOptions = options['Ureter']
+        ostiumDefaultOptions = ostiumOptions.getScaffoldSettings()
+        ostiumDefaultOptions['Ostium wall thickness'] = bladderWallThickness
+
+    @classmethod
+    def generateBaseMesh(cls, region, options):
+        '''
+        Generate the base tricubic Hermite mesh. See also generateMesh().
+        :param region: Zinc region to define model in. Must be empty.
+        :param options: Dict containing options. See getDefaultOptions().
+        :return: annotationGroups
+        '''
+        cls.updateSubScaffoldOptions(options)
+        elementsCountAlong = options['Number of elements along']
+        elementsCountAround = options['Number of elements around']
+        elementsCountThroughWall = options['Number of elements through wall']
+        majorDiameter = options['Major diameter']
+        minorDiameter = options['Minor diameter']
+        neckDiameter1 = options['Neck diameter 1']
+        neckDiameter2 = options['Neck diameter 2']
+        bladderWallThickness = options['Wall thickness']
+        useCrossDerivatives = options['Use cross derivatives']
+        useCubicHermiteThroughWall = not(options['Use linear through wall'])
+        neckAngleRadians = math.radians(options['Neck angle degrees'])
+
+        includeUrethra = options['Include urethra']
+        urethraDiameter1 = options['Urethra diameter 1']
+        urethraDiameter2 = options['Urethra diameter 2']
+        urethraWallThickness = options['Urethra wall thickness']
+        urethraproportion = options['Urethra length proportion to central path']
+
+        centralPath = options['Central path LUT']
+
+        includeUreter = options['Include ureter']
+        ostiumOptions = options['Ureter']
+        ostiumDefaultOptions = ostiumOptions.getScaffoldSettings()
+        elementsCountAroundOstium = ostiumDefaultOptions['Number of elements around ostium']
+        elementsCountAnnulusRadially = options['Number of elements radially on annulus']
+        ostiumPositionAround = options['Ostium position around']
+        ostiumPositionDown = options['Ostium position down']
+
+        firstNodeIdentifier = 1
+        firstElementIdentifier = 1
+
+        fm = region.getFieldmodule()
+        fm.beginChange()
+        nodes = fm.findNodesetByFieldDomainType(Field.DOMAIN_TYPE_NODES)
+        mesh = fm.findMeshByDimension(3)
+
+        # Central path
+        tmpRegion = region.createRegion()
+        centralPath.generate(tmpRegion)
+        cx, cd1, cd2, cd12 = extractPathParametersFromRegion(tmpRegion)
```

Comment:
This will be called twice to extract parameters for bladder and urethra separately, passing the group name. See comment earlier.

Reply:
Should I change the following part, too? Because the segmentLength will be different after deleting the "urethraLengthPortion". 

        # Find arcLength
        length = 0.0
        elementsCountIn = len(cx) - 1
        sd1 = interp.smoothCubicHermiteDerivativesLine(cx, cd1, fixAllDirections=True,
                                                       magnitudeScalingMode=interp.DerivativeScalingMode.HARMONIC_MEAN)
        for e in range(elementsCountIn):
            arcLength = interp.getCubicHermiteArcLength(cx[e], sd1[e], cx[e + 1], sd1[e + 1])
            length += arcLength
        segmentLength = length / elementsCountAlong

Reply:
Shouldn't need to smooth the central path. (Separately we need to find a way for the user to smooth the 1D path as needed; on the to-do list.)
You should have a separate cx, cd1 for bladder and urethra. Can have a fixed segment length for one or the other, but not both:
If it is the bladder, the first element of the urethra needs to transition from the derivatives supplied by the end of the bladder. If it is the urethra, the last element of the bladder needs to transition to fit the first urethra derivatives.